library/std/src/thread/local.rs

```diff
@@ -230,6 +230,14 @@ impl fmt::Display for AccessError {
 #[stable(feature = "thread_local_try_with", since = "1.26.0")]
 impl Error for AccessError {}
 
+// This ensures the panicking code is outlined from `with` for `LocalKey`.
+#[cfg_attr(not(feature = "panic_immediate_abort"), inline(never))]
+#[track_caller]
+#[cold]
+fn panic_access_error(err: AccessError) -> ! {
+    panic!("cannot access a Thread Local Storage value during or after destruction: {err:?}")
```

Comment:
Why did you add the `{err:?}` formatting? It's a ZST that only debug-prints its type name:
https://github.com/rust-lang/rust/blob/a580b5c379b4fca50dfe5afc0fc0ce00921e4e00/library/std/src/thread/local.rs#L217-L221

Reply:
I did it to mirror #115491
https://github.com/rust-lang/rust/blob/ad211ced81509462cdfe4c29ed10f97279a0acae/library/core/src/cell.rs#L799-L801
This code is not in the critical path, so I kept it closer to the current one; and if `AccessError` `Debug` implementation is changed in the future, it will be reflected here directly.

But I admit this is non necessary. I can inline it if you want.

Reply:
My reaction was more that "AccessError" is useless extra information to have at all, but now I realize that `Result::expect` was debug-printing that here before too. Whatever, we can leave it.